org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/junit/launcher/MockLaunch.java

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Erik Brangs and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Erik Brangs - initial implementation
+ *******************************************************************************/
+package org.eclipse.jdt.junit.launcher;
+
+import org.eclipse.debug.core.DebugException;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.model.IDebugTarget;
+import org.eclipse.debug.core.model.IProcess;
+import org.eclipse.debug.core.model.ISourceLocator;
+
+public class MockLaunch implements ILaunch {
+
+	private ILaunchConfiguration fILaunchConfiguration;
+
+	public MockLaunch(ILaunchConfiguration iLaunchConfiguration) {
+		fILaunchConfiguration= iLaunchConfiguration;
+	}
+
+	@Override
+	public ILaunchConfiguration getLaunchConfiguration() {
+		return fILaunchConfiguration;
+	}
+
+	// currently unneeded
+
+	@Override
+	public boolean canTerminate() {
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+	@Override
+	public boolean isTerminated() {
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+	@Override
+	public void terminate() throws DebugException {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public <T> T getAdapter(Class<T> adapter) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public Object[] getChildren() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public IDebugTarget getDebugTarget() {
+		// TODO Auto-generated method stub
```

Comment:
i have not looked into the details. but please do remove all "TODO"s
also there alsoready is a MockLaunch
org.eclipse.jdt.debug.tests.connectors.MockLaunch, please choose another name. Also it would be good to somehow have less boiler plate code - maybe extend Launch like NoopLaunch?

Reply:
Thank you. I've changed the code to use Launch which seems to work fine for the test cases.